racket/collects/racket/function.rkt

```diff
@@ -9,9 +9,12 @@
 
 (define (identity x) x)
 
-(define (const c)
-  (define (const . _) c)
```

Comment:
I think the original intention is to make it so that `(object-ame (const 1))` is `const`. So instead of returning `thunk*` directly, you might want to bind it to a variable `const` first, and then return it.

Reply:
`thunk*` seems to already be able to infer names:

```racket
Welcome to Racket v8.6 [cs].
> (object-name (const))
'const
> (object-name (const 1))
'const
> (object-name (const 1 1))
'const
```

Reply:
Oh nice. Nvm then. (I tested from the benchmark variant that you posted as a comment, and that variant doesn't work)